addon/-private/fields/attr.ts

```diff
@@ -1,24 +1,62 @@
-import { computed } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 import { Dict } from '@orbit/utils';
 
-export default function (type: string, options: Dict<unknown> = {}) {
-  options.type = type;
+import Model from '../model';
+
+export default function attr(type: string, options: Dict<unknown> = {}) {
+  function trackedAttr(target: any, key: string, desc: PropertyDescriptor) {
+    let trackedDesc = tracked(target, key, desc);
+    let { get: originalGet, set: originalSet } = trackedDesc;
+
+    let defaultAssigned = new WeakSet();
+
+    function setDefaultValue(record: Model) {
+      let value = record.getAttribute(key);
+      setValue(record, value);
+    }
+
+    function setValue(record: any, value: any) {
+      defaultAssigned.add(record);
+      return originalSet!.call(record, value);
+    }
+
+    function get(this: Model) {
+      if (!defaultAssigned.has(this)) {
+        setDefaultValue(this);
+      }
+      return originalGet!.call(this);
+    }
 
-  return computed({
-    get(key) {
-      return this.getAttribute(key);
-    },
-    set(key, value) {
+    function set(this: Model, value: any) {
       const oldValue = this.getAttribute(key);
 
       if (value !== oldValue) {
         this.replaceAttribute(key, value);
+        return setValue(this, value);
       }
 
       return value;
     }
-  }).meta({
-    options,
-    isAttribute: true
-  });
+
+    trackedDesc.get = get;
+    trackedDesc.set = set;
+
+    Reflect.defineMetadata('orbit:attribute', options, target, key);
+    Reflect.defineMetadata(
+      'orbit:notifier',
+      (record: Model) => setDefaultValue(record),
+      target,
+      key
+    );
+
+    return trackedDesc;
+  }
+
+  if (arguments.length === 3) {
```

Comment:
What is the scenario in which this condition is needed?

Reply:
It is the cas of `@attr() name;` vs `@attr name;`